org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/configuration/BuildConfiguration.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core.configuration;
+
+/**
+ * Describes a configuration of a Gradle build.
+ *
+ * @author Donat Csikos
+ *
+ */
+public interface BuildConfiguration {
```

Comment:
I think this should be a public value object. We can handle the connection between workspace config, build config and run configurations behind the scenes. No extra public service necessary.

Reply:
How exactly should we create new configuration objects then? With the [current design](https://github.com/eclipse/buildship/issues/732), the configuration manager object is responsible for the object creation/loading/storing. Shall we expose the builder objects as separate classes?

Reply:
Yep: `BuildConfiguration.forRootDirectory().withGradleDistribution().build`

Reply:
Done.